### PR TITLE
Add shared documents page

### PR DESCRIPTION
**Pull request by Mathieu414**
New page, restricted to coach/staff/admin which goal is to display administrative documents.

### CODE REVIEW

File: pages/shared_documents/shared_documents.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+require_once "utils/form_validation.php";
+
+$id = $_SESSION['user_id'];
+$post = $_POST;
+
+create_shared_docs_table();
+
+
+$v2 = validate($post);
+$file_upload = $v2->upload("file_upload")->set_target_dir("uploads/shared_docs/");
```
[the remark this answers is not in the document]
Comment (Atmos4) [reply pages/shared_documents/shared_documents.php:14]: I don't think this is necessary. We can trust people to name files properly :)

File: database/shared_docs.api.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+function set_shared_file($path, $date, $size, $mime)
+{
+    $result = query_db(
+        "INSERT INTO shared_documents(path, date, size, mime) VALUES (?,?,?,?) 
+        ON DUPLICATE KEY UPDATE path = ?, date= ?, size = ?, mime = ?;",
+        $path,
+        $date,
+        $size,
```
[the remark this answers is not in the document]
Comment (Atmos4) [reply database/shared_docs.api.php:23]: Maybe it can be good to check if the file path is already present, and throw an error if the file already there?

**Comment by Atmos4** [comment]
Otherwise LGTM, I can approve as soon as you address the comments!

File: database/shared_docs.api.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (Atmos4) [reply database/shared_docs.api.php:23]: What will happen now if the file already exists is that the file won't be created but a DB entry will be added, so maybe you need to check for file existence too

File: database/shared_docs.api.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<?php
+
+function create_shared_docs_table()
```
[the remark this answers is not in the document]
Comment (Mathieu414) [reply database/shared_docs.api.php:3]: I agree, this is a temporary solution to make it work. Will solve that in the future !! 💪

**Review by Atmos4 — APPROVED** [review]

**Pull request merged**